### ISSUE TITLE
Convergence of the learning curve and nan issue

### ISSUE THREAD

**Issue by BoomSky0416** [nie-lang/UnsupervisedDeepImageStitching#23]
To implement this method into low overlap scenes, I finetuned the provided 100,0000-checkpoint on new dataset for low overlap scenes. But the loss curve doesn't converge. And If I train the model from scratch on the new dataset, the nan issue happened. Here are the learning curves for Finetune and From scratch. Do you have any methods to fix it? Is it related to the quality or size of new dataset?
![image](https://user-images.githubusercontent.com/57380777/158503680-b0e6dcfb-b36c-4c68-94d8-bd5d1a2b2989.png)
![image](https://user-images.githubusercontent.com/57380777/158503768-830a9a43-6cab-42ab-bea0-7abec5c19578.png)

**Comment by nie-lang** [comment]
Since this unsupervised work is based on a pixel-level alignment loss, the low overlap would significantly increase the difficulty of training such a network.

In my view, the new dataset should not only contain scenes of low overlap. It's suggested to include some scenes of higher overlap to balance the data distribution. Maybe, it can make the training process more stable. Or you can try a lower learning rate.

Btw, it seems the loss slightly decreased during your finetuning. Maybe you could try to train it on a synthetic dataset first. Then, you can fine-tune it on your own dataset.

**Comment by BoomSky0416** [comment]
Thanks!
